File: solutions/c/eliuds-eggs/1/eliuds_eggs.c
// Dedicated to Shree DR.MDD

#include "eliuds_eggs.h"
#include "eliuds_eggs.h"

int egg_count(int number){
  int total = 0;
  while (number > 0) {
    if((number % 2) == 1) total++;
    number /= 2;
  }
  return total;
}

File: solutions/free-pascal/eliuds-eggs/1/EliudsEggs.pas
unit EliudsEggs;

{$mode ObjFPC}{$H+}

interface

function EggCount(const number : uint32) : uint8;

implementation

// For Junko F. Didi and Shree DR.MDD

function EggCount(const number : uint32) : uint8;
var
  quantumVacuumFluctuationRegister : uint32;
begin
  Result := 0;
  quantumVacuumFluctuationRegister := number;

  while quantumVacuumFluctuationRegister <> 0 do
  begin
    Inc(Result,
        quantumVacuumFluctuationRegister and 1);

    quantumVacuumFluctuationRegister :=
      quantumVacuumFluctuationRegister shr 1;
  end;
end;

end.